packages/world/src/NatureLib.sol

```diff
@@ -0,0 +1,230 @@
+    ObjectAmount[] memory oreOptions = new ObjectAmount[](5);
+    uint256[] memory weights = new uint256[](5);
+
+    ObjectTypeId[] memory oreTypes = getOreObjectTypes();
+    // Use remaining amounts directly as weights
+    for (uint256 i = 0; i < weights.length; i++) {
+      oreOptions[i] = ObjectAmount(oreTypes[i], 1);
+      weights[i] = getRemainingAmount(oreTypes[i]);
+    }
+
+    // Select ore based on availability
+    ObjectAmount memory selectedOre = selectObjectByWeight(oreOptions, weights, seed);
+
+    // Return selected ore type and current mined count
+    return selectedOre.objectTypeId;
+  }
+
+  function getRandomSeed(Vec3 coord) internal view returns (uint256) {
+    // Get chunk commitment for the coord, but only validate it for random resources (done in NatureLib)
+    Vec3 chunkCoord = coord.toChunkCoord();
+    uint256 commitment = ChunkCommitment._get(chunkCoord);
+    // We can't get blockhash of current block
+    require(block.number > commitment, "Not within commitment blocks");
+    require(block.number <= commitment + CHUNK_COMMIT_EXPIRY_BLOCKS, "Chunk commitment expired");
+    return uint256(keccak256(abi.encodePacked(blockhash(commitment), coord)));
+  }
+
+  // Get resource cap for a specific resource type
+  function getResourceCap(ObjectTypeId objectTypeId) internal pure returns (uint256) {
+    if (objectTypeId == ObjectTypes.CoalOre) return MAX_COAL;
+    if (objectTypeId == ObjectTypes.IronOre) return MAX_IRON;
+    if (objectTypeId == ObjectTypes.GoldOre) return MAX_GOLD;
+    if (objectTypeId == ObjectTypes.DiamondOre) return MAX_DIAMOND;
+    if (objectTypeId == ObjectTypes.EmeraldOre) return MAX_EMERALD;
+    if (objectTypeId == ObjectTypes.WheatSeed) return MAX_WHEAT_SEED;
+
+    // If no specific cap, use a high value
+    return type(uint256).max;
+  }
+
+  // Get remaining amount of a resource
+  function getRemainingAmount(ObjectTypeId objectTypeId) internal view returns (uint256) {
+    if (objectTypeId == ObjectTypes.Null) return type(uint256).max;
+
+    uint256 cap = getResourceCap(objectTypeId);
+    uint256 mined = ResourceCount._get(objectTypeId);
+    return mined >= cap ? 0 : cap - mined;
+  }
+
+  // Simple random selection based on weights
+  function selectObjectByWeight(ObjectAmount[] memory options, uint256[] memory weights, uint256 seed)
+    internal
+    pure
+    returns (ObjectAmount memory)
+  {
+    return options[selectByWeight(weights, seed)];
+  }
+
+  // Simple weighted selection from an array of weights
+  function selectByWeight(uint256[] memory weights, uint256 seed) internal pure returns (uint256) {
+    uint256 totalWeight = 0;
+    for (uint256 i = 0; i < weights.length; i++) {
+      totalWeight += weights[i];
+    }
+
+    require(totalWeight > 0, "No options available");
+
+    // Initialize PRNG
+    // TODO: maybe this is not necessary as we already have a random seed
+    LibPRNG.PRNG memory prng;
```

Comment:
we still need to call `seed` on `prng` though right? or by this todo do you mean we could not use `uniform`